numberOfProvinces.js

```diff
@@ -0,0 +1,106 @@
+/*
+n cities
+
+a - b - c
+
+i: n x n matrix
+o: total number of provinces
+
+directly or indirectly
+
+Input: isConnected = [[1,1,0],[1,1,0],[0,0,1]]
+Output: 2
+
+   0 1 2
+0 [1,1,0],
+1 [1,1,0],
+2 [0,0,1]
+
+*/
+
+const { Queue } = require('../utils/queue');
+
+const directions = [
+    [1, 0],
+    [0, 1],
+    [-1, 0],
+    [0, -1],
+];
+
+const CONNECTED = 1;
+
+const getRowColString = (row, col) => `${row}, ${col}`;
+
+const isBound = (row, col, matrix) => {
```

Comment:
This name is misleading. I'd expect a bound to be anything on the border of the matrix (so row `0` or col `0` or row `numRows - 1` or col `numCols - 1`). This function does NOT tell me if the given position is a bound. Rather, it tells me if the given position is WITHIN the bounds. I'd name it something like `isInBounds` or `isWithinBoundaries`.